executor/src/witgen/jit/single_step_processor.rs

```diff
@@ -31,148 +31,43 @@ impl<'a, T: FieldElement> SingleStepProcessor<'a, T> {
-            // we could also choose a different variable to branch on.
-            let left_branch_code =
-                self.generate_code_for_branch(can_process.clone(), first_branch, complete.clone())?;
-            let right_branch_code =
-                self.generate_code_for_branch(can_process, second_branch, complete)?;
-            if left_branch_code == right_branch_code {
-                common_code.into_iter().chain(left_branch_code).collect()
-            } else {
-                common_code
-                    .into_iter()
-                    .chain(std::iter::once(Effect::Branch(
-                        condition,
-                        left_branch_code,
-                        right_branch_code,
-                    )))
-                    .collect()
-            }
-        };
-        Ok(code)
-    }
-
-    fn initialize_witgen(&self) -> WitgenInference<'a, T, NoEval> {
         // All witness columns in row 0 are known.
-        let known_variables = self.machine_parts.witnesses.iter().map(|id| {
-            Variable::Cell(Cell {
-                column_name: self.fixed_data.column_name(id).to_string(),
-                id: id.id,
-                row_offset: 0,
-            })
+        let known_variables = all_witnesses.iter().map(|&id| self.cell(id, 0));
+        // and we want to know the ones in the next row.
+        let requested_known = all_witnesses.iter().map(|&id| self.cell(id, 1));
+        let identities = self.machine_parts.identities.iter().map(|&id| {
+            let row_offset = if id.contains_next_ref() { 0 } else { 1 };
+            (id, row_offset)
         });
-        WitgenInference::new(self.fixed_data, NoEval, known_variables)
-    }
-
-    fn process_until_no_progress<CanProcess: CanProcessCall<T> + Clone>(
-        &self,
-        can_process: CanProcess,
-        witgen: &mut WitgenInference<'a, T, NoEval>,
-        complete: &mut HashSet<u64>,
-    ) {
-        let mut progress = true;
-        while progress {
-            progress = false;
-
-            // TODO At this point, we should call a function on `witgen`
-            // to propagate known concrete values across the identities
-            // to other known (but not concrete) variables.
-
-            for id in &self.machine_parts.identities {
-                if complete.contains(&id.id()) {
-                    continue;
-                }
-                // TODO this is wrong if intermediate columns are referenced.
-                let row_offset = if id.contains_next_ref() { 0 } else { 1 };
-                let result = witgen.process_identity(can_process.clone(), id, row_offset);
-                progress |= result.progress;
-                if result.complete {
-                    complete.insert(id.id());
-                }
-            }
-        }
+        let block_size = 1;
+        let witgen = WitgenInference::new(self.fixed_data, NoEval, known_variables);
+
+        let max_branch_depth = 6;
```

Comment:
Again, please turn into a constant. Does it need to be a variable to `Processor::new`? Would we pass a different value in different situations? Otherwise, the constant could be in the processor file.

Reply:
We could pass different values for different machines. For example, a VM could find the pc lookup and use number of instruction variables * 2 or something.